### DISCUSSION TITLE
How to set visibility to private on a collection

### DISCUSSION THREAD

**Discussion by codered009** [filamentphp/filament#6080] [General]
I'm using spatie laravel media library plugin in my form.
https://filamentphp.com/docs/2.x/spatie-laravel-media-library-plugin/form-components
When uploading a file to a collection how to a collection how do set the visibility of the file to private as I do in Fileupload.

FileUpload::make('attachment')
    ->disk('s3')
    ->directory('form-attachments')
    ->visibility('private')

**Comment by hientt53** [comment]
Have you found a solution yet? I'm also experiencing the same problem.

**Reply by codered009** [reply]
Not yet. Did some workarounds specific to my project but that will not be a solution. Still waiting!

**Reply by hientt53** [reply]
Can you share your solution?
I'm planning to use MediaHasBeenAdded to set newly uploaded media as public based on custom properties.

**Comment by mahdi1113** [comment]
hi you can use this i have this problem and fix with this
->addCustomHeaders(['ACL' => 'public-read'])
